diff --git a/node_modules/@pkgr/webpack/lib/config.js b/node_modules/@pkgr/webpack/lib/config.js
index 52af7f3..f1f543a 100644
--- a/node_modules/@pkgr/webpack/lib/config.js
+++ b/node_modules/@pkgr/webpack/lib/config.js
@@ -182,6 +182,7 @@ exports.default = ({ entry = 'src', outputDir = 'dist', externals, type, copies
         output: {
             filename: filenamePrefix + 'js',
             path: path_1.default.resolve(outputDir),
+            publicPath: '/',
         },
         module: {
             rules: [
